src/formatting/syntux/parser.rs

```diff
@@ -99,7 +100,15 @@ pub(crate) enum ParserError {
 
 impl<'a> Parser<'a> {
     pub(crate) fn submod_path_from_attr(attrs: &[ast::Attribute], path: &Path) -> Option<PathBuf> {
-        rustc_expand::module::submod_path_from_attr(attrs, path)
+        let path_string = first_attr_value_str_by_name(attrs, sym::path)?.as_str();
+        // On windows, the base path might have the form
+        // `\\?\foo\bar` in which case it does not tolerate
+        // mixed `/` and `\` separators, so canonicalize
+        // `/` to `\`.
+        #[cfg(windows)]
+        let path_string = path_string.replace("/", "\\");
+
+        Some(path.join(&*path_string))
     }
```

Comment:
There were some noteworthy upstream changes that switched to using `Session` instead of `ParseSess`, and one impact is that `submod_path_from_attr` in rustc_expand now requires a Session reference so that rustc can report unused attributes. We just need to get the path value here though, so there's really no need for all the overhead. As such I've refactored to this to just look at the attributes directly and remove the dep on this rustc_expand function